src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable max-lines */
+import React from 'react';
+import { View, ViewStyle } from 'react-native';
+import styled, { withTheme } from 'styled-components/native';
+import { IAvatarSizes } from '@naturacosmeticos/natds-styles';
+import IconAnonymous from '../../assets/images/anonymous.png';
+
+import {
+  Theme,
+  getColorOnPrimary,
+  getColorPrimary,
+  getSpacingBySize,
+  getAvatarBySize,
+} from '../../common/themeSelectors';
+
+export type AvatarSizes = keyof IAvatarSizes;
+export type AvatarColors = 'primary' | 'secondary' | 'default' | string;
+export type AvatarTypes = 'image' | 'letter' | 'anonymous' | string;
+
+
+export interface AvatarProps {
+  /**
+   * Divider variants `image' | 'letter' | 'anonymous`
+   */
+  type?: AvatarTypes;
+  /**
+   * The divider theme
+   */
+  theme: Theme;
+  /**
+   * Optional Value used com Letter type
+   */
+  text?: string;
+  /**
+   * Optional URL image used com Image type
+   */
+  imgSource?: any;
+  /**
+   * Optional Color
+   */
+  color?: AvatarColors;
+  /**
+   * Optional Size
+   */
+  size?: AvatarSizes;
+  /**
+   * Optional accessibilityLabel
+   */
+  accessibilityLabel?: string;
+  /**
+   * Optional accessibilityHint
+   */
+  accessibilityHint?: string;
+  /**
+   * Optional ID for testing
+   */
+  testID?: string;
+}
+
+const getAvatarFontSize = (theme, size) => getAvatarBySize(size, theme).fontSize;
+
+const getAvatarSize = (theme, size) => getAvatarBySize(size, theme).size;
+
+const getViewStyles = (color: AvatarColors, size: AvatarSizes, theme: Theme):ViewStyle => ({
+  alignItems: 'center',
+  backgroundColor: getColorPrimary(theme),
+  borderRadius: getAvatarSize(theme, size),
+  height: getAvatarSize(theme, size),
+  justifyContent: 'center',
+  width: getAvatarSize(theme, size),
+});
+
+
+interface AvatarImage {
+  size: AvatarSizes;
+  theme: Theme;
+  source: any;
+  type: AvatarTypes;
+}
+
+interface AvatarLetter {
+  accessibilityLabel?: string;
+  accessibilityHint?: string;
+  accessibilityRole?: string;
+  size: AvatarSizes;
+  theme: Theme;
+  type: AvatarTypes;
+}
+
+const AvatarImage = styled.Image<AvatarImage>(({ size, theme }) => ({
+  borderRadius: getSpacingBySize(size, theme),
+  height: getAvatarSize(theme, size),
+  width: getAvatarSize(theme, size),
+}));
+
+const AvatarLetter = styled.Text<AvatarLetter>`
+  color: ${({ theme }) => getColorOnPrimary(theme)};
+  font-size: ${({ size, theme }) => getAvatarFontSize(theme, size)};
+  align-self: center;
+  letter-spacing: 1px;
+`;
+
+/**
+ * Get the first character from first and last word
+ * @param text string
+ */
+const getTextValue = (text) => {
+  if (text.length < 3) {
+    return text.toUpperCase();
+  }
+
+  // extract the first and last "first char" word
+  const firstLetters = text
+    .trim()
+    .split(' ')
+    .map(item => item[0]);
```

Comment:
Eu prefiro usar `string.charAt(0)` em vez do `string[0]`, porque aí se a string for vazia não corre o risco de retornar `undefined`. Acho que do jeito que está não quebra, mas o seguro morreu de velho. haha